mlrun/runtimes/databricks_job/databricks_wrapper.py

```diff
@@ -199,14 +205,20 @@ def print_status(run: Run):
                 cluster_id=cluster_id,
                 is_finished=True,
             )
-        run_output = workspace.jobs.get_run_output(get_task(run).run_id)
+
+        task_run_id = get_task(run).run_id
+        run_output = workspace.jobs.get_run_output(task_run_id)
     finally:
         workspace.dbfs.delete(script_path_on_dbfs)
         workspace.dbfs.delete(artifact_json_path)
 
+    run_result_state = run_output.metadata.state.result_state
+    if run_result_state == RunResultState.CANCELED:
```

Comment:
why this condition is needed?

Reply:
The bug was that the run state moved from aborting to completed and then to aborted when it should have moved from aborting straight to aborted.
we need to raise an exception if the task was aborted so that it won't be marked as completed/error.
I check that the task was aborted by checking the result state - if it is canceled - it was aborted